### PR TITLE
Simulation Function

### LINKED ISSUE #4: Run SImulation Function
We will need a function which runs the simulation of the game.  It should do the following

- take in the map
- each turn, make the player shoot at the closest zombie
- next, each zombie, (looping over top left to bottom left to keep it deterministic), will move 1 step foward towards the player
- if a zombie moves into the player, end the game
- if a zombie tries to move onto a rock, it'll find the next path path to get to the player
- if player kills all zombies, end the game
- if a zombie tries to move into a box, damage the box
- a box has 2 health
- remove the box from the map when box is destroyed
- zombie have 2 health

### PR DESCRIPTION
**Pull request by delasy**
Player shoots at the closest Zombie, algorithm I used (file lib/closest-entity.ts):
```
abs(
  (zombie.x + zombie.y) - (player.x + player.y)
)
```
No idea if that's the best algorithm for this, that was the first idea I came up with.

Wherever you place player/zombie, zombie will find a way to the player. There might be some edge cases I didn't test. Algorithm used - Breadth-first search: file lib/pathfinder.ts

Kill behavior: Zombie hits player, player disappears from the board and game ends (this is how visually it will look like).

Architecture decisions:
- multi games support, in case later we will add more games
- all entities Box, Rock, Player, Zombie inherit from Entity class
- pathfinder re-uses ZombieSurvival class to create different states to find the best solution for next Zombie move
- performance can be improved I guess, right now it's ~6ms per turn per zombie
- there's "snapshot" functionality. The way it works - it takes in config and converts it into a string by joining rows by period and cells by empty string.
- game/index.ts re-exports everything related to the game except lib functions
- from original issue, changed only one thing: Box has only 1HP (discussed on Discord).

Example how to use:
```ts
const game = new ZombieSurvival([
    ["P", " "],
    [" ", "Z"],
  ]);

while (!game.finished()) {
  game.step();
  // state is the same as config
  const state = game.getState();
}
```

Closes #4

### CODE REVIEW

File: games/ZombieSurvival/Direction.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export const allDirections = [
+  Direction.Down,
+  Direction.Left,
+  Direction.Right,
+  Direction.Up,
+];
+
+export function directionToString(direction: Direction): string {
```
Comment (delasy) [inline games/ZombieSurvival/Direction.ts:17]: `toString` and `fromString` needed by pathfinder, not used anywhere else
